rest/config_database.go

```diff
@@ -0,0 +1,16 @@
+package rest
+
+import (
+	"github.com/couchbase/sync_gateway/db"
+)
+
+// DatabaseConfig is a 3.x/persisted database config.
+// TODO: Review whether DatabaseConfig should maintain its own list of valid config options, or should just continue inheriting them from DbConfig
+type DatabaseConfig struct {
+	// cas is the Couchbase Server CAS of the database config in the bucket
+	cas uint64
+
+	Guest *db.PrincipalConfig `json:"guest,omitempty"`
```

Comment:
Why is Guest here, as opposed to in DbConfig?  I overlooked earlier that we're using DatabaseConfig (and not DbConfig) for the REST payload.  Is there an intentional distinction?

Reply:
The `DatabaseConfig` struct is currently a bit muddled - in that it stores both 3.x only config options, and fields persisted to the bucket but not returned by REST APIs.

I think we will need to revisit this anyway to strip out now invalid 2.x config options inside DbConfig (by removing the embed completely)